taker/src/main.rs

```diff
@@ -92,6 +92,11 @@ struct Opts {
     #[clap(long)]
     pub tokio_console: bool,
 
+    /// If enabled, libp2p ping and offer broadcast spans will be included in the traces exported
+    /// by the application.
+    #[clap(long)]
+    pub noisy_libp2p_spans: bool,
```

Comment:
I would keep the name a bit more general, so that we can use this toggle with other verbose traces if needed

```suggestion
    pub verbose_spans: bool,
```